pylint/testutils/_primer/primer_run_command.py

```diff
@@ -23,57 +22,48 @@
 class RunCommand(PrimerCommand):
     def run(self) -> None:
         packages: dict[str, list[OldJsonExport]] = {}
-        astroid_errors: list[Message] = []
-        other_fatal_msgs: list[Message] = []
+        fatal_msgs: list[Message] = []
         for package, data in self.packages.items():
-            messages, p_astroid_errors, p_other_fatal_msgs = self._lint_package(
-                package, data
-            )
-            astroid_errors += p_astroid_errors
-            other_fatal_msgs += p_other_fatal_msgs
+            messages, p_fatal_msgs = self._lint_package(package, data)
+            fatal_msgs += p_fatal_msgs
             packages[package] = messages
-        plural = "s" if len(other_fatal_msgs) > 1 else ""
-        assert not other_fatal_msgs, (
-            f"We encountered {len(other_fatal_msgs)} fatal error message{plural}"
-            " that can't be attributed to bleeding edge astroid alone (see log)."
-        )
```

Comment:
Just checking my understanding: if a fatal error happens on a main run that has nothing to do with `astroid-error`, we no longer fail the job (we just warn in a comment)? I wonder if we should still fail for non-astroid errors.

Reply:
Yeah. See my comment below. Relying on `astroid-error` clearly doesn't distinguish between "astroid bleeding edge" and "faulty PR code" enough as my PR caused an `astroid-error` without an astroid bump.
Thus, this gives a false sense of security which should hopefully be removed by now failing CIs on PR completely.

Reply:
Yeah, I get that. I have no problem with the PR case. I'm wondering about the `main` case. Incompatible PRs are merged, or a new commit is pushed to a primer project, or some indeterminacy somewhere causes something to fail on main that we never noticed in a PR. Now, we can have failures on main with green CI. No?

Reply:
Yeah, but I think we "sort of" already could. The only `fatal` messages that are not `astroid-error` are:
`method-check-failed`, `parse-error`, `config-parse-error` and `fatal`.

1. I know `fatal` can only be raised in very specific circumstances and does not normally occur while linting a fail as there we raise `astroid-error` instead of `fatal` then.
2. `method-check-failed` doesn't have any tests so might me moot?
3. `parse-error` is raised when `astroid` can't import a module, so that's essentially a different `astroid-error`.
4. `config-parse-error` is related to configuration parsing and not very likely to trigger.

The problem is that `astroid-error` is our main error for anything causing a crash during linting. The other four are only raised in special circumstances. We weren't really catching fatal errors on `main` anyway 😅 
I think it makes more sense to find a good way to distinguish `astroid-errors` caused by bleeding edge or by a merged PR and act according to it.

Reply:
Or a way to stop reraising any failures stemming from pure pylint as `AstroidError`, but that's backwards incompatible.